Business.Tests/CommentServiceTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿using AutoFixture;
+using AutoMapper;
+using Business.DataTransferObjects;
+using Business.Exceptions;
+using Business.Services;
+using Data.Entities;
+using Data.Interfaces;
+using ExpectedObjects;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Business.Tests;
+
+public class CommentServiceTests
+{
+    private readonly CommentService _sut;
+    private readonly Mock<IUnitOfWork> _unitOfWorkMock = new();
+
+    private readonly Fixture _fixture = new();
+    private readonly IMapper _mapper = UnitTestHelper.CreateMapper();
+
+    public CommentServiceTests()
+    {
+        _sut = new CommentService(_unitOfWorkMock.Object, _mapper);
+        _fixture.Behaviors.Remove(new ThrowingRecursionBehavior());
+        _fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+    }
+
+    [Fact]
+    public async Task CreateAsync_ShouldCreateGame_WhenParentCommentUnspecified()
+    {
+        var commentCreationDto = _fixture.Build<CommentCreationDto>()
+            .Without(c => c.ParentId)
+            .Create();
+        var mappedComment = _mapper.Map<Comment>(commentCreationDto);
+        var expectedToCreate = mappedComment.ToExpectedObject();
+        var expected = _mapper.Map<CommentDto>(mappedComment);
+
+        _unitOfWorkMock.Setup(u => u.GameRepository.GetByIdAsync(commentCreationDto.GameId))
+            .ReturnsAsync(_fixture.Create<Game>());
+        _unitOfWorkMock.Setup(u => u.CommentRepository.Add(It.IsAny<Comment>()));
+
+        var result = await _sut.CreateAsync(commentCreationDto);
+
+        result.Should().BeEquivalentTo(expected);
+
+        _unitOfWorkMock.Verify(u => u.CommentRepository.Add(It.Is<Comment>(g => expectedToCreate.Equals(g))),
```

Comment:
You can just do `u.CommentRepository.Add(expectedToCreate)`. Delegate for more complicated rules

Reply:
I can't do this because expectedToCreate is of type ExpectedObject

Reply:
You're right